### ISSUE TITLE
Non realURL urls rendered/generated

### ISSUE THREAD

**Issue by jonathanheilmann** [jonathanheilmann/ext-jh_opengraphprotocol#51]
If the first and cached page-request is done without a realURL url, the query is cached and used for any further request.
It's preferred to generate a realURL url instead

**Comment by jonathanheilmann** [comment]
This should solve the issue, proposed in PR #49 by @olegsv.
@olegsv could you please test branch v1.3.0 and leave some feedback?

- [ ] This fix should be tested within a TYPO3 CMS installation using CoolURI

**Comment by olegsv** [comment]
@jonathanheilmann I've checked. v1.3.0 produces correct URL for my setup ( RealURL )
However if I specify page via its pid , i.e. index.php?id=34 , the og:url still contains RealURL url . Not sure whether this is correct or incorrect behavior.

**Comment by jonathanheilmann** [comment]
That's the behavior I intended. As readable URLs are better rated in SEO aspects, these URLs should be used.
It's a question of design: The original way is to use the URL of the first call after flushing cache. This could be a readable one, or that one using the non-readable query. The solution I've committed will always generate a readable URL (if supported through RealURL or CoolURI) and save it in cache.

Maybe it's an option to provide a checkbox in extension configuration to enable forced generation of readable URLs?
